languages/thingtalk/ast_manip.ts

```diff
@@ -1467,54 +1401,50 @@ function addGetPredicateJoin(table : Ast.Expression,
         return addFilterInternal(table, newAtom, {});
     }
 
-    let newAtom = new Ast.BooleanExpression.Atom(null, 'id', (lhsArg.type.isArray ? 'in_array' : '=='), new Ast.Value.VarRef(lhsArg.name));
-    if (negate)
-        newAtom = new Ast.BooleanExpression.Not(null, newAtom);
-    const get_predicate = new Ast.BooleanExpression.External(null,
-        get_predicate_table.expression.invocation.selector,
-        get_predicate_table.expression.invocation.channel,
-        get_predicate_table.expression.invocation.in_params,
-        new Ast.BooleanExpression.And(null, [get_predicate_table.filter, newAtom]),
-        get_predicate_table.expression.invocation.schema
+    // add id projection to subquery
+    if (subquery instanceof Ast.ProjectionExpression)
+        subquery.args = ['id'];
```

Comment:
Do you need to clone subquery before you modify it like this?

Reply:
what issue will it cause without clone?

Reply:
The same object that you pass in, used in a different template, will be modified and cause havoc.